lib/Doctrine/DBAL/Driver/Mysqli/MysqliStatement.php

```diff
@@ -230,6 +246,27 @@ public function execute($params = null)
         return true;
     }
 
+    /**
+     * Send BLOB type streams using mysqli::send_long_data
+     */
+    private function _sendLongData()
+    {
+        foreach ($this->_longData as $paramNr => $resource) {
+            if (is_resource($resource) && get_resource_type($resource) === 'stream') {
+                while (! feof($resource)) {
+                    $longData = fread($resource, 8192);
+                    if (! $this->_stmt->send_long_data($paramNr, $longData)) {
+                        throw new MysqliException($this->_stmt->error, $this->_stmt->sqlstate, $this->_stmt->errno);
+                    }
+                }
+            } else {
+                if (! $this->_stmt->send_long_data($paramNr, $resource)) {
```

Comment:
Why use `send_long_data()` for non-streams? We should bind them as usual.

Reply:
I could change that like so: https://gist.github.com/mpdude/1ad1c03556d9fd4191396159b7929124

Basically, if you bind as `ParameterType::LARGE_OBJECT` but do *not* pass in a stream, just treat it like `ParameterType::STRING`. Might lose the advantage of sending the parameter in a network package by itself, though.

Reply:
Yeah, valid point. Let's send strings via `send_long_data()` too. Please rename `$resource` to `$value` then, since it's not always a resource and it's by design.

Reply:
:+1:

Reply:
> Might lose the advantage of sending the parameter in a network package by itself, though.

This did make sense since I thought one could send a string which exceeds `max_allowed_packet` using `send_long_data()`. Given it's not the case, what's the advantage of sending the parameter in a network package by itself? Isn't it better to just bind the string as is and let the driver decide how to send it?